src/utils/localstorage.ts

```diff
@@ -39,6 +39,30 @@ export function getLocalstorageWithSchema<Schema extends z.ZodTypeAny>(
   return out!;
 }
 
+export function getNullableLocalstorageWithSchema<Schema extends z.ZodTypeAny>(
```

Comment:
Maybe we should remove the check from `getLocalstorageWithSchema` and just pass `undefined` to the schema, so something like (untested)

```ts
export function getLocalstorageWithSchema<Schema extends z.ZodTypeAny>(
  key: string,
  schema: Schema,
): z.infer<Schema> {
  let value: unknown;
  const valueRaw = localStorage.getItem(key);
  if (valueRaw !== null) {
    try {
      value = JSON.parse(valueRaw);
    } catch (err) {
      throw new ErrorInvalidJson(key);
    }
  } else {
    value = undefined;
  }
  const results = schema.safeParse(value);
  if (!results.success) {
    throw results.error;
  }
  const out = results.data as z.infer<Schema>;
  return out!;
}
```

The we can allow the schema to accept `null` as a value

Reply:
Yeah maybe . I just didn't want to rock the boat especially as this is all temporary until we get our user data api from planning squad